### ISSUE TITLE
Upgrading from 1.4.0 to 1.5.0 kill mssql extension

### ISSUE THREAD

**Issue by jeansagi** [Microsoft/vscode-mssql#1200]
- MSSQL Extension Version: 1.5.0
- VSCode Version: 1.32.3
- OS Version: Windows 10 Pro 1809 x64

Steps to Reproduce:

1. Open any file with SQL extension
2. After a few seconds in Output Console Tasks view shows the message:
```
Unhandled Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.SqlTools.Utility.Logger.Close() in D:\a\1\s\src\Microsoft.SqlTools.Hosting\Utility\Logger.cs:line 79
   at Microsoft.SqlTools.ServiceLayer.Program.Main(String[] args) in D:\a\1\s\src\Microsoft.SqlTools.ServiceLayer\Program.cs:line 27
[Error - 17:26:10] Connection to server got closed. Server will not be restarted.

```

I had Version 1.4.0 working beautifully via TCP connection and upgrading to 1.5.0 just kill mssql extension.

Please fix this ASAP.

Any information you need, do not hesitate to ask for it.

BTW: If I uninstall 1.5.0 how do I install 1.4.0?

J.

**Comment by jeansagi** [comment]
Uninstalled version 1.5.0 and reinstalled it. The same problem arises.

But before the unhandled exception, it shows the following messages to the console:
```
Initializing SQL tools service for the mssql extension.
Note: mssql commands will be available after installing the service.

Platform: win32, x86_64 (Windows)

Installing SQL tools service to C:\Users\Jean\.vscode\extensions\ms-mssql.mssql-1.5.0\sqltoolsservice\1.5.0-alpha.79\Windows.
Downloading https://github.com/Microsoft/sqltoolsservice/releases/download/v1.5.0-alpha.79/microsoft.sqltools.servicelayer-win-x64-netcoreapp2.2.zip
(42950 KB) .................... Done!
Installing ...
```

And just a few seconds after "the Installing..." message the unhandled exception message appears.

So it seems to me that there is a problem initializing the SQL tools service.
I looked at the installation directory and it is just a bunch of .exe, .dll and .json files. there is no log or any other file that shows any indication of the error.

J.

**Comment by jeansagi** [comment]
For installing a previous version you goto to the extension and right click it. The choose the option "Install another version". I chose 1.4.0 and Code did the rest... Very nice BTW.

Just to mention that this time the extension got installed without any problem:
```
Initializing SQL tools service for the mssql extension.
Note: mssql commands will be available after installing the service.

Platform: win32, x86_64 (Windows)

Installing SQL tools service to C:\Users\Jean\.vscode\extensions\ms-mssql.mssql-1.4.0\sqltoolsservice\1.4.0\Windows.
Downloading https://download.microsoft.com/download/3/2/9/32995640-2B4A-4C47-8764-287485C733DB/microsoft.sqltools.servicelayer-win-x64-netcoreapp2.1.zip
(37396 KB) .................... Done!
Installing ...
Done! 413 files unpacked.
```

Maybe it is related to the netcoreapp2.2?
Or downloading from an alpha?
Or perhaps it is related to [#1194 Refresh Intellisense cache option don't work](https://github.com/Microsoft/vscode-mssql/issues/1194), cause after installing 1.4.0 Intellisense got broke...

Anyway, with 1.4.0 mssql returned to normal (almost)...

J.

**Comment by kburtram** [comment]
@jeansagi the 1.5 update was primarily to fix #1194, which caused Intellisense to no longer work with the VS Code 1.32.3+ releases.  So downgrading to 1.4 will break Intellisense.

Could you please check if you have this setting `"mssql.logDebugInfo": true`?  If so, could you please try setting to `false` or removing this setting?  If you don't have this setting enabled, could you please share which `mssql` setting you do have set?

I've only been able to repro with this setting, and will work on a fix for that now.

**Comment by jeansagi** [comment]
Hi, @kburtram thanks for your support.

I have indeed the setting you mention set to true ( "mssql.logDebugInfo": true ).

I set the "mssql.logDebugInfo": false in Settings but all is behaving the same... I'm not sure how setting a log to false will help..., in fact, I'm lost, I mean I don't understand what are you trying to fix... Intellisense on 1.4.0 after reverting from 1.5.0 or fixing the broke of 1.5.0 after upgrading from 1.4.0 .

FWIW here are the mssql settings I've set:
```
    //-------- mssql Configuration --------
    // [Optional] Log debug output to the VS Code console (Help -> Toggle Developer Tools)
    "mssql.logDebugInfo": false,
    // The maximum number of recently used connections to store in the connection list.
    "mssql.maxRecentConnections": 10,
    "mssql.messagesDefaultOpen": false,
    "mssql.connections": [
        {
            "server": "tcp:<prdctionSrver1>\\<Instnce1>,50596",
            "database": "<dtabse1>",
            "authenticationType": "SqlLogin",
            "user": "user1",
            "password": "",
            "savePassword": true,
            "profileName": "INSTNCE1"
        },
        {
            "server": "tcp:<prdctionSrvr2>,60597",
            "database": "dtbse2",
            "authenticationType": "SqlLogin",
            "user": "user2",
            "password": "",
            "savePassword": true,
            "profileName": "PrdctionSrvr2"
        }
    ],
"mssql.resultsFontSize": 10
```

J.

**Comment by kburtram** [comment]
@jeansagi sorry for not being more clear.  I'm investigating the null reference unhandled exception issue you encountered when upgrading to v1.5.0.  The null reference appears to be triggered by having `logDebugInfo` set to `true`.

I've pushed this fix into the v1.5.0 extension https://github.com/Microsoft/sqltoolsservice/pull/788.  You should now be able to update to the latest extension version to pick-up the IntelliSense fix without encountering the null reference issue.  Please let me know if you continue to encounter issues with the v.1.5.0 extension.

**Comment by jeansagi** [comment]
Hi @kburtram,

Well, indeed it seems it *"was"* related to *"logDebugInfo"*.

I uninstalled 1.4.0 and installed 1.5.0, while logDebugInfo was set to false. Now the installation completed successfully!:
```
Initializing SQL tools service for the mssql extension.
Note: mssql commands will be available after installing the service.

Platform: win32, x86_64 (Windows)

Installing SQL tools service to c:\Users\Jean\.vscode\extensions\ms-mssql.mssql-1.5.0\sqltoolsservice\1.5.0-alpha.79\Windows.
Downloading https://github.com/Microsoft/sqltoolsservice/releases/download/v1.5.0-alpha.79/microsoft.sqltools.servicelayer-win-x64-netcoreapp2.2.zip
(42952 KB) .................... Done!
Installing ...
Done! 437 files unpacked.
```

After that, I've set logDebugInfo to **_true_** and **_false_** while cycling *"Code"* and it works without any hitch, even Intellisense!!.

So I'm closing this issue now.

Thank you really, very much. 
mssql is the reason for me to use Code, it saves me a lot of time daily.


J.
